pkg/resource/plugin/provider_plugin.go

```diff
@@ -260,30 +261,53 @@ func (p *provider) Create(urn resource.URN, props resource.PropertyMap) (resourc
 		return "", nil, resource.StatusOK, err
 	}
 
+	var id resource.ID
+	var properties *_struct.Struct
+	var resourceError *rpcerror.Error
+	var resourceStatus = resource.StatusOK
 	resp, err := client.Create(p.ctx.Request(), &pulumirpc.CreateRequest{
 		Urn:        string(urn),
 		Properties: mprops,
 	})
 	if err != nil {
-		resourceStatus, rpcErr := resourceStateAndError(err)
-		logging.V(7).Infof("%s failed: err=%v", label, rpcErr)
-		return "", nil, resourceStatus, rpcErr
+		resourceStatus, resourceError = resourceStateAndError(err)
+		contract.Assert(resourceError != nil)
+		logging.V(7).Infof("%s failed: err=%v", label, resourceError)
+
+		for _, detail := range resourceError.Details() {
+			// If resource was successfully created but failed to initialize, the error will be packed
+			// with the live properties of the object.
+			if initErr, ok := detail.(*pulumirpc.ErrorResourceInitFailed); ok {
+				id = resource.ID(initErr.GetId())
+				properties = initErr.GetProperties()
+				resourceStatus = resource.StatusPartialFailure
```

Comment:
Nit: `break`

I might even pull the extraction of the id, properties, and status from `resp, err` out into its own function (especially given that you need something very similar for `Update`).